x/protocolpool/keeper/keeper.go

```diff
@@ -0,0 +1,226 @@
+	// Get current balance of the intermediary module account
+	moduleAccount := k.authKeeper.GetModuleAccount(ctx, types.ProtocolPoolEscrowAccount)
+	if moduleAccount == nil {
+		return errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", types.ProtocolPoolEscrowAccount)
+	}
+	params, err := k.Params.Get(ctx)
+	if err != nil {
+		return err
+	}
+
+	// only take into account the balances of denoms whitelisted in EnabledDistributionDenoms
+	amountToDistribute := sdk.NewCoins()
+	for _, denom := range params.EnabledDistributionDenoms {
+		bal := k.bankKeeper.GetBalance(ctx, moduleAccount.GetAddress(), denom)
+		amountToDistribute = append(amountToDistribute, bal)
+	}
+
+	// if the balance is zero, return early
+	if amountToDistribute.IsZero() {
+		return nil
+	}
+
+	remainingCoins := sdk.NewCoins(amountToDistribute...)
+	iter, err := k.ContinuousFunds.Iterate(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create iterator for continuous funds: %w", err)
+	}
+
+	kvalues, err := iter.KeyValues()
+	if err != nil {
+		return fmt.Errorf("failed to iterate continuous funds: %w", err)
+	}
+
+	blockTime := ctx.BlockTime()
+	anyNegative := false
+	for _, kv := range kvalues {
+		recipient := kv.Key
+		fund := kv.Value
+
+		// remove newly expired funds
+		if fund.Expiry != nil && fund.Expiry.Before(blockTime) {
+			err := k.ContinuousFunds.Remove(ctx, recipient)
+			if err != nil {
+				return fmt.Errorf("failed to remove fund for %s from ContinuousFunds: %w", recipient, err)
+			}
+			continue
+		}
+
+		amountToStream := PercentageCoinMul(fund.Percentage, amountToDistribute)
+		remainingCoins, anyNegative = remainingCoins.SafeSub(amountToStream...)
+		if anyNegative {
+			return fmt.Errorf("negative funds for distribution from ContinuousFunds: %v", remainingCoins)
+		}
+
+		err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ProtocolPoolEscrowAccount, recipient, amountToStream)
+		if err != nil {
+			return fmt.Errorf("failed to distribute fund for %s from ContinuousFunds: %w", recipient, err)
+		}
+	}
+
+	// send all remaining funds to the community pool
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ProtocolPoolEscrowAccount, types.ModuleName, remainingCoins); err != nil {
+		return fmt.Errorf("failed to send coins to community pool: %w", err)
+	}
+
+	return nil
+}
+
+// GetAllContinuousFunds gets all continuous funds in the store.
+func (k Keeper) GetAllContinuousFunds(ctx sdk.Context) ([]types.ContinuousFund, error) {
+	var cf []types.ContinuousFund
+	err := k.ContinuousFunds.Walk(ctx, nil, func(key sdk.AccAddress, value types.ContinuousFund) (stop bool, err error) {
+		recipient, err := k.authKeeper.AddressCodec().BytesToString(key)
+		if err != nil {
+			return true, err
+		}
+		cf = append(cf, types.ContinuousFund{
+			Recipient:  recipient,
+			Percentage: value.Percentage,
+			Expiry:     value.Expiry,
+		})
+		return false, nil
+	})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
it, err := k.ContinuousFunds.Iterate(ctx, nil)
cf, err := it.Values()

Reply:
fixed